File: CS/04_Graphs/DrawDashedLine.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.ColorSpace;
using Spire.Pdf.Graphics;

namespace DrawDashedLine
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Load a PDF file
            PdfDocument pdf = new PdfDocument();
            pdf.LoadFromFile(@"..\..\..\..\..\..\Data\DrawingTemplate.pdf");

            //Get the first page
            PdfPageBase page = pdf.Pages[0];

            //Save graphics state
            PdfGraphicsState state = page.Canvas.Save();

            //Set location and size for line
            float x = 150;
            float y = 200;
            float width = 300;

            //Create pens and set syle for it
            PdfPen pen = new PdfPen(Color.Red, 3f);

            //Set dash style and pattern
            pen.DashStyle = PdfDashStyle.Dash;
            pen.DashPattern = new float[] { 1, 4, 1 };

            //Draw dashed lines
            page.Canvas.DrawLine(pen, x, y, x + width, y);

            //Restore graphics
            page.Canvas.Restore(state);

            String result = "DrawDashedLine_out.pdf";

            //Save the document
            pdf.SaveToFile(result);

            //Launch the Pdf file
            PDFDocumentViewer(result);
        }
        private void PDFDocumentViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
    
}

File: VB/04_Graphs/DrawDashedLine.vb
Imports Spire.Pdf
Imports Spire.Pdf.Graphics

Namespace DrawDashedLine
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new PDF document
            Dim pdf As New PdfDocument()

            ' Load an existing PDF from a file
            pdf.LoadFromFile("..\..\..\..\..\..\Data\DrawingTemplate.pdf")

            ' Get the first page of the loaded PDF
            Dim page As PdfPageBase = pdf.Pages(0)

            ' Save the current graphics state of the page's canvas
            Dim state As PdfGraphicsState = page.Canvas.Save()

            ' Set the starting point (x, y) and width for the dashed line
            Dim x As Single = 150
            Dim y As Single = 200
            Dim width As Single = 300

            ' Create a pen with color red and thickness 3f
            Dim pen As New PdfPen(Color.Red, 3.0F)

            ' Set the dash style of the pen to Dash
            pen.DashStyle = PdfDashStyle.Dash

            ' Set the dash pattern for the pen
            pen.DashPattern = New Single() {1, 4, 1}

            ' Draw a dashed line on the page using the pen, starting from (x, y) and extending to (x + width, y)
            page.Canvas.DrawLine(pen, x, y, x + width, y)

            ' Restore the graphics state of the page's canvas
            page.Canvas.Restore(state)

            ' Specify the output file name
            Dim result As String = "DrawDashedLine_out.pdf"

            ' Save the modified PDF document to the output file
            pdf.SaveToFile(result)

            ' Close the PDF document
            pdf.Close()

            ' Launch the Pdf file
            PDFDocumentViewer(result)
        End Sub
        Private Sub PDFDocumentViewer(ByVal filename As String)
            Try
                Process.Start(filename)
            Catch
            End Try
        End Sub
    End Class

End Namespace
